web/themes/custom/novel/novel.theme

```diff
@@ -276,3 +277,43 @@ function novel_theme_suggestions_select_alter(array &$suggestions, array $variab
     $suggestions[] = 'select__single';
   }
 }
+
+/**
+ * Preprocesses rows in a Drupal view to handle the display of event instances.
+ *
+ * This function iterates over each row in the view, checking for the presence
+ * of an EventInstance object.
+ * When an EventInstance is found, it checks if it belongs to the same
+ * event series as the previous row.
+ * If so, it marks the current row as 'stacked' and adjusts its view
+ * mode accordingly.
+ * This is useful for grouping events from the same series in the view,
+ * enhancing visual continuity and clarity.
+ *
+ * @param array &$variables
+ *   An array of variables to pass to the theme layer.
+ */
+function novel_preprocess_views_view_unformatted(&$variables): void {
+  $rows = &$variables['rows'];
+
+  foreach ($rows as $index => &$row) {
+    if (isset($row['content']['#eventinstance'])) {
+      /** @var \Drupal\recurring_events\Entity\EventInstance $eventInstance */
+      $eventInstance = $row['content']['#eventinstance'];
+
+      if ($eventInstance instanceof EventInstance) {
+        $eventSeriesId = $eventInstance->getEventSeries()->id();
+
+        // Check if not the first row and if event_series_id matches
+        // the previous row.
+        $isStacked = $index > 0 && $eventSeriesId == $rows[$index - 1]['content']['#eventinstance']->getEventSeries()->id();
```

Comment:
```suggestion
      $eventInstance = $row['content']['#eventinstance'];
      $previousInstance = ($index > 0) ? $rows[$index - 1]['content']['#eventinstance'] : NULL;
      if ($eventInstance instanceof EventInstance && $previousInstance instanceof EventInstance) {
        $isStacked = $eventInstance->getEventSeries()->getId() == $previousInstance->getEventSeries()->id();
```

Consider restructuring the code. In my opinion this version above (totally untested!) is preferable:

1. It adds proper type checking for the previous row
2. It eliminates unnecessary variables
3. It is shorter without sacrificing clarity

Reply:
There is something about this suggested change that breaks the code 😅

Reply:
Without more info it is hard to tell whats wrong. I guess we will just have to leave it as is.